### DISCUSSION TITLE
Will there be an update on the periodic & parallel at some point?

### DISCUSSION THREAD

**Discussion by pagitrog** [usnistgov/fipy#906] [Q&A]
Dear FiPy community,

I am working with a phase-field model and the Cahn-Hilliard equation to predict equilibrium shapes of biological lipid membranes. For such problems, systems are necessarily 3D and require periodic BCs, and because of the system size parallelization of the computational domain is crucial. So, I am interested in employing FiPy for parallelized simulations of periodic 3D systems. 

My question is probably redundant as there is an open issue on the bugged parallelization of periodic systems (https://github.com/usnistgov/fipy/issues/518). I nevertheless would like to ask about this, as the last posted requests and updates are from 2019. If I missed an important aspect, please let me know. I can also provide example code.

My findings principally are that, it is not possible to parallelize any of the 3D periodic grid options. All options drop the same error (the same for the latest but different FiPy versions). I however found that the "PeriodicGrid2DLeftRight" can be parallelized, and a quick visual inspection looks promising. However, the phase-fields are not-conserved as one anticipates when using Cahn-Hilliard. This is probably related to a number of posted issues on .globalValue (e.g. https://github.com/usnistgov/fipy/issues/856).
Otherwise, using FiPy has been a great experience!

**Comment by guyer** [comment]
Thanks for the query. Periodic & parallel is high on my queue to address; we need it for some benchmarking exercises we're doing. I realize that issue has been open for awhile, but I actually have good reason to believe that I'll be able to devote more time to FiPy development than for the past several years.

Based on how we partition the domain in `PeriodicGrid2DLeftRight`, I would expect it to parallelize. Issue #856 is about `FaceVariable` objects. There's no reason to expect those to be conservative. You should be evaluating conservation with a `CellVariable`. If you've got a test case that doesn't conserve, I'd appreciate seeing it.
